jbpm-wb-human-tasks/jbpm-wb-human-tasks-client/src/main/java/org/jbpm/workbench/ht/client/editors/taskslist/AbstractTaskListPresenter.java

```diff
@@ -130,6 +134,23 @@ protected DataSetReadyCallback getDataSetReadyCallback(final Integer startRange,
                 .withEmptyResultsCallback(() -> setEmptyResults());
     }
 
+    private Set<Long> findDuplicateId(LinkedList<TaskSummary> taskSummaries) {
+        Set<Long> id = new HashSet<>();
+        return taskSummaries.stream().filter(ts -> !id.add(ts.getId())).map(TaskSummary::getId).collect(Collectors.toSet());
+    }
+
+    protected void removeDuplicateTaskSummaryBySLAStatus(LinkedList<TaskSummary> taskSummaries) {
+        Set<Long> duplicateTaskSummaries = findDuplicateId(taskSummaries);
+        if (!duplicateTaskSummaries.isEmpty()) {
+            for (int i = 0; i < taskSummaries.size(); i++) {
+                Integer slaPending = new Integer(ProcessInstance.SLA_PENDING);
+                if (duplicateTaskSummaries.contains(taskSummaries.get(i).getId()) && slaPending.equals(taskSummaries.get(i).getSlaCompliance())) {
+                    taskSummaries.remove(taskSummaries.get(i));
```

Comment:
I think removing any taskSummaries  item inside the loop over the same taskSummaries could be problematic.

Reply:
I do not understand.

Reply:
What I mean It's regarding that ConcurrentModificationException..When you remove one item from the list you are parsing, the list changes, then the loop can be inconsistent.

Reply:
so I change to removeif